rollouts/controllers/remotesync_controller.go

```diff
@@ -175,15 +192,22 @@ func (r *RemoteSyncReconciler) syncExternalSync(ctx context.Context, rs *gitopsv
 		return "", fmt.Errorf("failed to create client: %w", err)
 	}
 
-	if err := r.patchExternalSync(ctx, dynCl, syncName, rs); err != nil {
+	var namespace string
+	if rs.Spec.Type == gitopsv1alpha1.TemplateTypeRepoSync {
+		namespace = rs.Namespace
+	} else {
+		namespace = rootSyncNamespace
+	}
+
+	if err := r.patchExternalSync(ctx, dynCl, syncName, rs, namespace); err != nil {
 		return "", fmt.Errorf("failed to create/update sync: %w", err)
 	}
 
 	r.setupWatches(ctx, rs.Name, rs.Namespace, rs.Spec.ClusterRef)
 
-	syncStatus, err := checkSyncStatus(ctx, dynCl, syncName)
+	syncStatus, err := checkSyncStatus(ctx, dynCl, syncName, rs.Spec.Type, namespace)
```

Comment:
nit: can we keep `namespace` and `name` together (they represent identity of the sync object)

Reply:
I am now passing the entire `remoterootsync` object to the function, so that it can directly access the type, name, and namespace.